kytos/core/buffers.py

```diff
@@ -150,9 +187,18 @@ def __init__(self):
         self.msg_in = KytosEventBuffer("msg_in",
                                        maxsize=self._get_maxsize("sb"),
                                        queue_cls=PriorityQueue)
-        self.msg_out = KytosEventBuffer("msg_out",
-                                        maxsize=self._get_maxsize("sb"),
-                                        queue_cls=PriorityQueue)
+        strategy = limits.strategies.MovingWindowRateLimiter(
+            limits.storage.MemoryStorage()
+        )
+        self.msg_out = RateLimitedBuffer(
+            "msg_out",
+            maxsize=self._get_maxsize("sb"),
+            queue_cls=PriorityQueue,
+            strategy=strategy,
+            limit=limits.RateLimitItemPerSecond(100, 1),
+            gen_identifiers=lambda event:
+                getattr(event.destination, 'id', ('unknown',)),
```

Comment:
Excellent to have the gen_indentifiers as `Callable[[KytosEvent], Iterable[Hashable]]`. This should work correctly for the same limit for every dpid, but if wanted a different rate limit per dpid that wouldn't be possible. I believe the most common use case most will have the same rate limit, but in the configuration we should support:

- No rate limit at all on `msg_out`
- Rate limit on `msg_for`. Once Rate limit is set in the config, then there should be an option to either use the same limit for any switch, or a more specific one. Could you come up with a configuration on kytos.conf for this? I have an impression that on kytos.conf it should suffice and not need to be completely dynamic with APIs. 

We can also consider having a default have limit by default to ensure it's safer by default, 100 FlowMods per sec by default seems very reasonable to me, since it's aligned with the value that mef_eline implements at the client side.

Reply:
Also, once you figure out a configuration for this, please update the changelog accordingly.

Reply:
Added in the configuration, plus updated the changelog.